coderd/tracing/status_writer.go

```diff
@@ -33,10 +33,11 @@ func StatusWriterMiddleware(next http.Handler) http.Handler {
 }
 
 func (w *StatusWriter) WriteHeader(status int) {
-	if !w.wroteHeader {
-		w.Status = status
-		w.wroteHeader = true
+	if w.wroteHeader {
+		return
 	}
```

Comment:
Can you add a log here to have the stacktrace of whoever is doing this? Because this shouldn't be happening

Reply:
Do you think I should pipe slog through here?

Reply:
hmmmm idk. It'd be great to see what causes these though

Reply:
Devil's advocate: if it shouldn't happen, then should it instead be a fatal error and cause the entire request to fail? We can probably only do this in unit tests and pass in a parameter `failOnDuplicateWrite = flag.Lookup("test.v")` or similar.

Reply:
Agreed. I'll fix this!

Reply:
Make sure it has the full stacktrace and not just the caller, because the caller will almost always be `httpapi.*`